### PR TITLE
[MERGE] website_slides: add an animation when user levels up

### PR DESCRIPTION
**Pull request by awa-odoo**
PURPOSE

If a user levels up after passing a quiz, we want to clearly show him in the success modal that he
actually passed a level instead of just showing his new rank and xp gauge.

This commit introduces an animation with the progress bar when the user is levelling up.

After reaching the upper bound of the next level:
- The progress bar goes back to 0 ;
- The lower/upper bounds change to reflect the new reached level bounds ;
- The message "Amazing!" becomes "Level Up!" ;
- The motivational phrase changes with a fade out/fade in animation.

LINKS

PR : #39421
Task ID : 2032649

### CODE REVIEW

File: addons/website_slides/static/src/scss/website_slides.scss [hunk 1]
```diff
@@ -29,6 +29,29 @@ $o-wslides-fs-side-width: 300px;
     }
 }
 
+.progress {
```
[the remark this answers is not in the document]
Comment (awa-odoo) [reply addons/website_slides/static/src/scss/website_slides.scss]: I missed that while reviewing, good catch!

File: addons/website_slides/static/src/xml/slide_quiz.xml [hunk 2]
```diff
@@ -96,24 +96,39 @@
...
                             <div>
-                                <h1 class="mt-3 display-4 font-weight-bold">Amazing!</h1>
+                                <h1 class="o_wslides_quiz_modal_title mt-3 display-4 font-weight-bold">Amazing!</h1>
                                 <div class="pb-3">
-                                    <h4 class="pb-2 d-flex">You gained <span class="badge badge-pill badge-success text-white font-weight-bold ml-2 mr-1"><t t-esc="widget.quiz.quizKarmaWon"/> XP</span> !</h4>
-                                    <div class="progress" style="height: 8px">
-                                        <div class="progress-bar" role="progressbar" t-att-aria-valuenow="widget.quiz.rankProgress.progress" aria-valuemin="0" aria-valuemax="100"
-                                            t-attf-style="width: #{widget.quiz.rankProgress.progress}%"/>
+                                    <h4 class="o_wslides_xp_gained pb-2 d-flex fade">You gained <span class="badge badge-pill badge-success text-white font-weight-bold ml-2 mr-1"><t t-esc="widget.quiz.quizKarmaWon"/> XP</span> !</h4>
```
[the remark this answers is not in the document]
Comment (awa-odoo) [reply addons/website_slides/static/src/xml/slide_quiz.xml]: Classes are now namespaced with "o_wslides_quiz_modal" prefix, thanks for the suggestion.

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-ff

File: addons/website_slides/views/website_slides_templates_profile.xml [hunk 3]
```diff
@@ -58,9 +56,9 @@
 
                             <div class="d-flex align-items-center">
                                 <div class="progress flex-grow-1" style="height:0.5em">
-                                    <div class="progress-bar bg-primary" t-att-style="'width: '+ str(channel_completion)+'%'"/>
+                                    <div class="progress-bar bg-primary" t-att-style="'width: '+ str(course.completion)+'%'"/>
```
Comment (xmo-odoo) [inline addons/website_slides/views/website_slides_templates_profile.xml]: Could use t-attf-style to make this a bit more readable

Comment (awa-odoo) [reply addons/website_slides/views/website_slides_templates_profile.xml]: Indeed, but it's a little late now (I just noticed your comment).

Comment (tde-banana-odoo) [reply addons/website_slides/views/website_slides_templates_profile.xml]: You should be more readable.

**Comment by KangOl** [comment]
So this PR should be r+ed

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**